Microsoft.Toolkit/Collections/ObservableGroup.cs

```diff
@@ -49,9 +52,21 @@ public ObservableGroup(TKey key, IEnumerable<TValue> collection)
         }
 
         /// <summary>
-        /// Gets the key of the group.
+        /// Gets or sets the key of the group.
         /// </summary>
-        public TKey Key { get; }
+        public TKey Key
+        {
+            get
+            {
+                return _key;
+            }
+
+            set
+            {
+                _key = value;
+                OnPropertyChanged(new PropertyChangedEventArgs(nameof(Key)));
+            }
```

Comment:
We'd probably want to keep the consistency with the `ObservableObject` class in the MVVM Toolkit with respect to when the `PropertyChanged` event is raised. Also some minor code style nits 😄
```suggestion
            set
            {
                if (!EqualityComparer<TKey>.Default.Equals(this.key, value))
                {
                    this.key = value;
                    
                    OnPropertyChanged(new PropertyChangedEventArgs(nameof(Key)));
                }
            }
```